src/hstr.c

```diff
@@ -1312,16 +1269,21 @@ void loop_to_select(Hstr *hstr)
             LOGSELECTION(Y_OFFSET_HELP,getmaxy(stdscr),hstr->selectionSize);
 
             if(c>K_CTRL_Z) {
-                selectionCursorPosition=SELECTION_CURSOR_IN_PROMPT;
-
                 if(strlen(pattern)<(width-basex-1)) {
                     strcat(pattern, (char*)(&c));
                     print_pattern(pattern, hstr->promptY, basex);
                     cursorX=getcurx(stdscr);
                     cursorY=getcury(stdscr);
                 }
 
-                result = hstr_print_selection(maxHistoryItems, pattern, hstr);
+                previousSelectionCursorPosition = SELECTION_CURSOR_IN_PROMPT;
+                result = hstr_make_and_print_selection(pattern, hstr);
+                if (hstr->selectionSize > 0) {
+                    selectionCursorPosition=0;
```

Comment:
I should revert this. I was thinking it might be nice to have the first matching entry always selected to be more explicit about what you get when you hit enter, but should be separated into another patch.